opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -2619,6 +2630,16 @@ namespace Opm {
         }
     }
 
+    template<typename TypeTag>
+    double
+    BlackoilWellModel<TypeTag>::
+    wellPI(const std::string& well_name) const
+    {
+        auto well_iter = std::find_if(this->wells_ecl_.begin(), this->wells_ecl_.end(), [&well_name](const Well& well) { return well.name() == well_name; });
+        auto well_index = std::distance( this->wells_ecl_.begin(), well_iter );
```

Comment:
Another time, but this time even with a check whether this is at the end. See above,

Reply:
I have added the check here; have not created a separate function as that would need to expose iterator type in the API -  and I am not really comfortable with that (yet anyway) ...